src/search.cpp

```diff
@@ -1473,7 +1488,7 @@ void RootMove::insert_pv_in_tt(Position& pos) {
   StateInfo state[MAX_PLY], *st = state;
   size_t idx = 0;
 
-  for ( ; idx < pv.size(); ++idx)
+  for ( ; idx < std::min(validPvLength, pv.size()); ++idx)
```

Comment:
Yes, this is required to be a non-functional change.  Otherwise we will insert more entries into the TT than before.  I tested that in the past, and it was a regression.